base_layer/mmr/src/sparse_merkle_tree/tree.rs

```diff
@@ -859,4 +869,50 @@ mod test {
         // Sanity check that the tree is now empty
         assert!(tree.is_empty());
     }
+
+    #[test]
+    fn insert_only_if_exist() {
+        let mut tree = SparseMerkleTree::<Blake2b<U32>>::default();
+
+        // An empty tree contains no keys
+        assert!(!tree.contains(&short_key(0)));
+        assert!(!tree.contains(&short_key(1)));
+
+        // Add a key, which the tree must then contain
+        assert_eq!(
+            tree.insert(short_key(1), ValueHash::from([1u8; 32])).unwrap(),
+            UpdateResult::Inserted
+        );
+        assert!(!tree.contains(&short_key(0)));
+        assert!(tree.contains(&short_key(1)));
+        assert_eq!(
+            tree.insert(short_key(1), ValueHash::from([1u8; 32])).unwrap(),
+            UpdateResult::AlreadyExistingKey
+        );
```

Comment:
```suggestion
        assert_eq!(
            tree.insert(short_key(1), ValueHash::from([2u8; 32])).unwrap(),
            UpdateResult::AlreadyExistingKey
        );
        assert_eq!(tree.get(&short_key(1)).unwrap().unwrap(), &ValueHash::from([1u8; 32]));
```
This asserts that the value is in fact unchanged, and that an upsert did not occur.